crates/sui-bridge/src/lib.rs

```diff
@@ -30,10 +30,13 @@ pub(crate) mod test_utils;
 
 // TODO: can we log the error very time it gets retried?
 #[macro_export]
-macro_rules! retry_with_max_delay {
-    ($func:expr, $max_delay:expr) => {{
-        let retry_strategy = ExponentialBackoff::from_millis(100)
-            .max_delay($max_delay)
+macro_rules! retry_with_max_attempt_times {
+    ($func:expr, $max_retry_times:expr) => {{
+        // The following delay sequence will be used, applied with jitter
+        // 0.04, 0.08, 0.16, 0.32, 0.64, 1.28, 2.56, 5.12, 10.24, 20.48, 30.0, 30.0, 30.0 ...
+        let retry_strategy = ExponentialBackoff::from_millis(4)
```

Comment:
Did you mean 40?

Reply:
I meant 4 but my math was dead .... fixing

Reply:
`ExponentialBackoff` in tokio_retry is not very ergonomic. I'm going to switch to `backoff` crate